server/models/shares.js

```diff
@@ -451,7 +451,7 @@ async function regenerateRoleNamesTable() {
 
 
 function throwPermissionDenied() {
-    throw new interoperableErrors.PermissionDeniedError(tMark('permissionDenied'));
+    throw new interoperableErrors.PermissionDeniedError(tUI('permissionDenied', config.defaultLanguage));
```

Comment:
I don't know how to pass the current user language selected in locale param

Reply:
I'm afraid, it is not supposed to work. The exceptions are not supposed to be localized since they are not the way we communicate with end-users. Exception is and should be the point when we or the administrator failed (e.g. in setting the permissions). 

The tMark is used to mark the point where the translation is used in the code so that the automated script (extract.js) can identify it and transform it into string resource id (e.g. 'permissionDenied').
The tUI is used for generated translated strings on the server. In this case, however, the language has to be given as a parameter because the server does not assume language implicitly.

Reply:
We use https://github.com/smhg/express-locale that add to req the language, perhaps we can extract it from request.

Reply:
You Are right that we could possibly extract it from the cookie sent together with the REST request. However, it does not really change much on the fact that "Permission denied" is not something the end-user should see. The UI should not offer choices the user has not permissions for. If the permission denied can be obtained, then the client should catch the exception and show it in a more user-friendly way. As such, I would really not localize the exceptions. Exceptions are for logs, not for end-users.

Reply:
Can we do the translation in flash notice only?
![imagen](https://user-images.githubusercontent.com/1895563/91654119-a613af00-eaa6-11ea-97c8-52995cec50a2.png)

Reply:
Sure. Only it should not be translated by the server, but by the client.

Check client/src/lib/modals.js:164. I belive you could add something similar to general error handling in client/src/lib/page.js:461. There I would check for the `error instanceof interoperableErrors.PermissionDeniedError` and would present a fitting error message possibly with giving some hints (like someone may have changed the permissions while you worked, if you think it should work, contact the administrator or similar).